pkg/proxy/proxy.go

```diff
@@ -159,27 +159,44 @@ func (p *Proxy) health(ctx echo.Context) error {
 func (p *Proxy) processRequest(ctx echo.Context) (string, *access.ClusterAccess, error) {
 	userID, _ := ctx.Get(context.SubKey).(string)
 	username, _ := ctx.Get(context.UsernameKey).(string)
-	proxyPluginName, workspace, err := getWorkspaceContext(ctx.Request())
+	proxyPluginName, workspaceName, err := getWorkspaceContext(ctx.Request())
 	if err != nil {
 		return "", nil, crterrors.NewBadRequest("unable to get workspace context", err.Error())
 	}
 
-	ctx.Set(context.WorkspaceKey, workspace) // set workspace context for logging
-	cluster, err := p.app.MemberClusterService().GetClusterAccess(userID, username, workspace, proxyPluginName)
+	ctx.Set(context.WorkspaceKey, workspaceName) // set workspace context for logging
+	cluster, err := p.app.MemberClusterService().GetClusterAccess(userID, username, workspaceName, proxyPluginName)
```

Comment:
Good point. I've moved the cluster retrieval to the end in https://github.com/codeready-toolchain/registration-service/pull/371/commits/e936a0b2c8b6403172e2a54ae6f36956115aa26d

But I haven't done the simplification of `GetClusterAccess` yet, since that will require some unit test updates, let's do that in the next PR if that is ok for you. 

Thanks for the suggestion.

Reply:
Never mind I have reverted this change in https://github.com/codeready-toolchain/registration-service/pull/371/commits/138336d1cefde5b7b318b0282b76461c064e25b0 

Since the tests needed to be updated even just for moving the function down. So let's address this separately if that sounds good.